kitchen/src/kitchen/core.clj

```diff
@@ -511,9 +520,12 @@
           (when (pos? start-up-sleep-ms)
             (log/info "sleeping for" start-up-sleep-ms "ms before starting server")
             (Thread/sleep start-up-sleep-ms))
-          (let [partial-server-options {:ring-handler (basic-auth-middleware username password (params/wrap-params http-handler))
+          (let [partial-server-options {:ring-handler (->> (params/wrap-params http-handler)
+                                                           (basic-auth-middleware username password)
+                                                           correlation-id-middleware)
```

Comment:
Going to polish it (add `x-cid` in response in another PR) in kitchen and then change Waiter too. That way we can also ensure the presence of the `x-cid` header in all our responses.